apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -136,7 +170,7 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
   async function handleCreateClick(deposit: number, depositorAddress: string) {
     setIsCreatingDeployment(true);
 
-    const sdl = selectedSdlEditMode === "yaml" ? editedManifest : (sdlBuilderRef.current?.getSdl() as string);
+    const sdl = selectedSdlEditMode === "yaml" && editedManifest ? editedManifest : (sdlBuilderRef.current?.getSdl() as string);
```

Comment:
The create click should be disabled if the `editedManifest` is not defined. This will get the sdl from the sdl builder if it's not defined while being in the yaml mode.

Reply:
const sdl = selectedSdlEditMode === "yaml" ? editedManifest as string : (sdlBuilderRef.current?.getSdl() as string);
Is this what you want?